src/server/api/routers/post.ts

```diff
@@ -46,7 +48,23 @@ export const postRouter = createTRPCRouter({
             // query: z.function().args().returns(z.string())
         }))
         .query(async ({input}) => {
-            const queryURI = encodeURI(input.query);
+            console.log("SEARCH: " + input.query)
+
+            const openai = new OpenAI({
+                apiKey: env.OPENAI_API_SECRET
+            })
+
+            const chatCompletion = await openai.chat.completions.create({
+                model: "gpt-4-turbo-preview",
+                messages: [{"role": "system", "content": `
+                You will be given text. Find the opposite text to the prompt.
+
+                For example, If the input is 'Hot locations near me', response should be 'Cold locations far away from me'
```

Comment:
Would it be possible to add more examples? I fear this will only produce decent results in a few cases. 

Providing the input as JSON and requesting a JSON output can also be helpful, as this reduces the risk of receiving “Sure! Here is ...” or “Certainly!”.

Also, we currently don't support location information, so I'm not sure how this example would work.